### PR TITLE
Add additional metadata parameters to OpenIdConnectConfiguration

### LINKED ISSUE #2498: [Feature Request] Add some missing metadata parameters to OpenIdConnectConfiguration
Some metadata parameters are missing in `OpenIdConnectConfiguration`:

- `prompt_values_supported` from [Initiating User Registration via OpenID Connect 1.0](https://openid.net/specs/openid-connect-prompt-create-1_0.html)
- `pushed_authorization_request_endpoint` and `require_pushed_authorization_requests` from [OAuth 2.0 Pushed Authorization Requests](https://datatracker.ietf.org/doc/html/rfc9126)
- `backchannel_authentication_endpoint`, `backchannel_token_delivery_modes_supported`, `backchannel_authentication_request_signing_alg_values_supported` and `backchannel_user_code_parameter_supported` from [OpenID Connect Client-Initiated Backchannel Authentication Flow](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html)
- `dpop_signing_alg_values_supported` from [OAuth 2.0 Demonstrating Proof of Possession (DPoP)](https://datatracker.ietf.org/doc/html/rfc9449)
- `authorization_response_iss_parameter_supported` from [OAuth 2.0 Authorization Server Issuer Identification](https://datatracker.ietf.org/doc/html/rfc9207)

### PR DESCRIPTION
**Pull request by ghost**
Fixes #2498.

- [x] You've read the [Contributor Guide](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/Contributing.md) and [Code of Conduct](https://opensource.microsoft.com/codeofconduct/).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [ ] If any gains or losses in performance are possible, you've included benchmarks for your changes. [More info](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/wiki/Benchmarking-in-Wilson)

### CODE REVIEW

**Comment by halter73** [comment]
@brentschmaltz Do you think this will make it in soon? It'd be nice if we could use these new APIs in https://github.com/dotnet/aspnetcore/issues/51686.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs [hunk 1]
```diff
@@ -46,6 +46,10 @@ public class OpenIdConnectConfiguration : BaseConfiguration
         private ICollection<string> _userInfoEndpointEncryptionAlgValuesSupported;
         private ICollection<string> _userInfoEndpointEncryptionEncValuesSupported;
         private ICollection<string> _userInfoEndpointSigningAlgValuesSupported;
+        private ICollection<string> _promptValuesSupported;
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/Configuration/OpenIdConnectConfiguration.cs]: @joegoldman2  please put these in lexicographic order.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs [hunk 2]
```diff
@@ -58,6 +58,15 @@ public static class OpenIdProviderMetadataNames
         public const string UserInfoEncryptionAlgValuesSupported = "userinfo_encryption_alg_values_supported";
         public const string UserInfoEncryptionEncValuesSupported = "userinfo_encryption_enc_values_supported";
         public const string UserInfoSigningAlgValuesSupported = "userinfo_signing_alg_values_supported";
+        public const string PromptValuesSupported = "prompt_values_supported";
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs]: @joegoldman2 please keep in lexicographic order.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs [hunk 3]
```diff
@@ -58,6 +58,15 @@ public static class OpenIdProviderMetadataNames
...
+        public const string PushedAuthorizationRequestEndpoint = "pushed_authorization_request_endpoint";
+        public const string RequirePushedAuthorizationRequests = "require_pushed_authorization_requests";
+        public const string BackchannelAuthenticationEndpoint = "backchannel_authentication_endpoint";
+        public const string BackchannelTokenDeliveryModesSupported = "backchannel_token_delivery_modes_supported";
+        public const string BackchannelAuthenticationRequestSigningAlgValuesSupported = "backchannel_authentication_request_signing_alg_values_supported";
+        public const string BackchannelUserCodeParameterSupported = "backchannel_user_code_parameter_supported";
+        public const string DPoPSigningAlgValuesSupported = "dpop_signing_alg_values_supported";
+        public const string AuthorizationResponseIssParameterSupported = "authorization_response_iss_parameter_supported";
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs]: @joegoldman2 please keep in lexicographic order.

File: src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs [hunk 4]
```diff
@@ -114,5 +123,14 @@ internal static class OpenIdProviderMetadataUtf8Bytes
         public static ReadOnlySpan<byte> UserInfoEncryptionAlgValuesSupported => "userinfo_encryption_alg_values_supported"u8;
         public static ReadOnlySpan<byte> UserInfoEncryptionEncValuesSupported => "userinfo_encryption_enc_values_supported"u8;
         public static ReadOnlySpan<byte> UserInfoSigningAlgValuesSupported => "userinfo_signing_alg_values_supported"u8;
+        public static ReadOnlySpan<byte> PromptValuesSupported => "prompt_values_supported"u8;
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdProviderMetadataNames.cs]: @joegoldman2 please keep in lexicographic order.

File: test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMetadata.json [hunk 5]
```diff
@@ -1,46 +1,55 @@
...
-    "userinfo_encryption_enc_values_supported" : ["A256CBC-HS512", "A128CBC-HS256"],
-    "userinfo_signing_alg_values_supported" : ["ES384", "ES512"]
+    "token_endpoint_auth_methods_supported": ["client_secret_post", "private_key_jwt"],
+    "token_endpoint_auth_signing_alg_values_supported": ["ES192", "ES256"],
+    "ui_locales_supported": ["hak-CN", "en-us"],
+    "userinfo_endpoint": "https://login.microsoftonline.com/add29489-7269-41f4-8841-b63c95564420/openid/userinfo",
+    "userinfo_encryption_alg_values_supported": ["ECDH-ES+A128KW","ECDH-ES+A192KW"],
+    "userinfo_encryption_enc_values_supported": ["A256CBC-HS512", "A128CBC-HS256"],
+    "userinfo_signing_alg_values_supported": ["ES384", "ES512"],
+    "prompt_values_supported": ["none", "login", "consent"],
```
Comment (brentschmaltz) [inline test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMetadata.json]: @joegoldman2 please keep in lexicographic order.

**Comment by joegoldman2** [comment]
Will update the PR for the lexicographic order a bit later today or tomorrow

**Comment by joegoldman2** [comment]
@brentschmaltz I reorganized the code to keep the lexicographic order. Do you think we add the missing params from https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/2456 as well?

**Comment by brentschmaltz** [comment]
@joegoldman2 Thanks for the reordering.
Let's handle #2456 separately.

**Review by brentschmaltz — CHANGES_REQUESTED** [review]
I don't see corresponding changes to OpenIdConnectConfigurationSerializer.

File: test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMetadata.json (hunk 5 above)
Comment (brentschmaltz) [reply test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/OpenIdConnectMetadata.json]: @halter73 we need to add to the OpenIdConnectConfigurationSerializer.

**Comment by ghost** [comment]
> OpenIdConnectConfigurationSerializer

What do you mean? OpenIdConnectConfigurationSerializer has been updated a well in this PR.

**Comment by brentschmaltz** [comment]
@joegoldman2 ahh, the file was collapsed and i missed it.
I am looking it over.

**Review by brentschmaltz — APPROVED** [review]

**Comment by brentschmaltz** [comment]
@joegoldman2 approved.

**Comment by josephdecock** [comment]
I think the milestone on this issue should be 7.6.1. It looks like that's when this was actually released?

**Pull request merged**